### PR TITLE
[openstack/utils] Topology refinement

### PR DESCRIPTION
**Pull request by fwiesel**
- Use default instead of value as the default value from global is not properly merged
- Add service annotations for topology aware routing
- Add the annotations only in regions with more than one az

### CODE REVIEW

**Comment by fwiesel** [comment]
Follow up to #4965

**Comment by Carthaca** [comment]
With removing the `.global.topology_key`value, can you please also rework `kubernetes_pod_az_affinity`? 
I noticed, that the `.Values` are anyhow not working in https://github.com/sapcc/helm-charts/pull/4965/files#diff-a8000139f34581224d5ac4276723eb108ae9879555d815bbe05577f0076d4534R35, the input to the function is only a availability zone name

**Comment by fwiesel** [comment]
Done: I've renamed it to `utils.kubernetes_pod_az_affinity`, and pass it `envAll`. I'll need then to update the charts using it when making use of it.

File: openstack/utils/templates/_affinity.tpl [hunk 1]
```diff
@@ -47,3 +48,4 @@ affinity:
                   values:
                   - operational
 {{- end }}
+s
```
Comment (Carthaca) [inline openstack/utils/templates/_affinity.tpl:51]: This `s` comes probably unintended

**Review by Carthaca — CHANGES_REQUESTED** [review]
Thanks, a tiny extra letter sneaked in :)

File: openstack/utils/templates/_affinity.tpl (hunk 1 above)
Comment (fwiesel) [reply openstack/utils/templates/_affinity.tpl]: Sorry, indeed. I've removed it.

**Review by Carthaca — APPROVED** [review]
Thanks!

**Pull request merged**